syntax = "proto3";

option java_package = "ex.grpc";
option objc_class_prefix = "HSW";

package example;

service BiddingWriteService {
  rpc ListProduct (ListRequest) returns (ListReply) {}
  rpc PlaceBid (BidRequest) returns (BidReply) {}
}

message ListRequest {
  string product_name = 1;
  int32 seller_id = 2;
  double price = 3;
}

message ListReply {
  string message = 1;
}

message BidRequest {
  int32 product_id = 1;
  int32 bidder_id = 2;
  double price = 3;
}

message BidReply {
  string message = 1;
}

